### DISCUSSION TITLE
NextCloud AiO Multiple Istances + NGINX Proxy Manager = problems with Talk app

### DISCUSSION THREAD

**Discussion by DrPleb** [nextcloud/all-in-one#2517] [Questions] [answered]
Hi, I'm trying to deploy multiple istances of NextCloud AiO and as the GitHub post says "The normal way is creating multiple VMs, installing AIO in [reverse proxy mode](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md) in each of them and having one reverse proxy in front of them that points to each VM (you also need to use a different TALK_PORT for each of them). So, i'm using VMware to deploy multiple VM of NextCloud AiO, [nc01.domain.com](https://nc01.domain.com/), [nc02.domain.com](https://nc02.domain.com/), ecc...in reverse proxy mode and another VM for NGINX Proxy Manager. Everything works except Talk app, it keeps restarting the call when you are doing it, and in the settings it says there is a problem with STUN/TURN serverI have two theories:- in NGINX Proxy Manager, I used Streams on 1002 port (I change the default TALK_PORT to this one), so i'm forwarding requests that arrives on the proxy manager, that is hosting my domain [nc01.domain.com](https://nc01.domain.com/), to the ip address of my nc01 istance, TCP and UDP...and maybe for some reason isn't working (?)- I'm not understanding what is the meaning of "multiple VM with one reverse proxy in front of them"...maybe I have to install NGINX Proxy Manager on the same machine that is hosting NextCloud? But if I do this, what is the purpose of the reverse proxy? Having it or not having it is the same thing, it is all in the same vm and also I have to reserve a public IP address for each nextcloud aio istance...

Here is my NGINX Proxy Manager docker-compose.yml:

```
version: '3.8'

services:

app:

image: 'jc21/nginx-proxy-manager:latest'

restart: unless-stopped

ports:

# These ports are in format <host-port>:<container-port>

- '80:80' # Public HTTP Port

- '443:443' # Public HTTPS Port

- '81:81' # Admin Web Port

- '1002:1002/tcp' #NextCloud talk port

- '1002:1002/udp' #NextCloud talk port

environment:

# Mysql/Maria connection parameters:

DB_MYSQL_HOST: "db"

DB_MYSQL_PORT: 3306

DB_MYSQL_USER: "npm"

DB_MYSQL_PASSWORD: "npm"

DB_MYSQL_NAME: "npm"

# Uncomment this if IPv6 is not enabled on your host

DISABLE_IPV6: 'true'

volumes:

- ./data:/data

- ./letsencrypt:/etc/letsencrypt

depends_on:

- db



db:

image: 'jc21/mariadb-aria:latest'

restart: unless-stopped

environment:

MYSQL_ROOT_PASSWORD: 'npm'

MYSQL_DATABASE: 'npm'

MYSQL_USER: 'npm'

MYSQL_PASSWORD: 'npm'

volumes:

- ./mysql:/var/lib/mysql

```

Here is my docker-run command:

```
sudo docker run \

--sig-proxy=false \

--name nextcloud-aio-mastercontainer \

--restart always \

--publish 8080:8080 \

--env APACHE_PORT=1001 \

--env TALK_PORT=1002 \

--env APACHE_IP_BINDING=[0.0.0.0](https://0.0.0.0/) \

--env NEXTCLOUD_DATADIR="/nc_data" \

--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \

--volume /var/run/docker.sock:/var/run/docker.sock:ro \

nextcloud/all-in-one:latest

```

Can you help me understand this? Thank you guys <3

**Comment by szaimen** [comment, marked as answer]
Hi, in general things look correct but the Talk port is not allowed to be proxied. It must be directly forwarded to each VM that is why you need to choose a different talk port for each VM and open that in your router udp and tcp. You can check this also in the talk admin settings for each instance.

Edit: apparently a port over 1024 was required.

**Reply by DrPleb** [reply]
Hi, thanks for your answer, so you mean that I need to have only one public IP address, a custom port for talk (1002 for istance nc01, 1004 for istance nc02, ecc...) and the configuration should work...without using NGINX Proxy Manager for Talk. So I have only to change settings in my firewall.

Thank you, I'll keep you updated

**Reply by DrPleb** [reply]
Sorry, but I can't get this working, I tried this.
wan ip: 7.6.5.4
nc01: 10.10.10.11
nginx proxy manager: 10.10.10.10
talk port: 1002

NPM works on public and forward correctly to my nextcloud https...
on my firewall if you check 7.6.5.4:1002 t redirects correctly to 10.10.10.11:1002.
In Nextcloud admin page, I can see that talk is setup like this nc01.domain.com:1002 and if you check the internally and externally it works correctly, I can see it also from firewall logs that the requests passes, but I keep seeing this red signal in nextcloud talk admin page.

I tried a new istance without reverse proxy and it works correctly.
![Immagine 2023-05-09 093721](https://user-images.githubusercontent.com/132892195/237027235-45656a54-34b8-432e-80d9-35310800cf9b.png)

**Reply by DrPleb** [reply]
I tried with the Nextcloud AiO in reverse proxy mode WITHOUT changing the port, so keep the 3478/TCP and 3478/UDP, and it works. If I change the port with "--env TALK_PORT=1002" and obviously changing it also in the firewall it doesn't work. I also tried to reinstall the VM.
Here you can see that if I change the port, on the container details I still see the 3478 port...maybe is it a bug?
![Immagine 2023-05-09 133929](https://github.com/nextcloud/all-in-one/assets/132892195/be3cee0c-ee4c-45bf-a58a-03461860ec36)

**Reply by szaimen** [reply]
Possibly the setting is not adjusted in the talk admin settings? Can you check if the port matches the chosen talk port and if adjusting that makes it work?

**Reply by DrPleb** [reply]
It matches, in the Talk admin settings the Talk port is the one that I choose during installation and it correctly changes...I don't know how to debug this

**Reply by szaimen** [reply]
Can you post the output of `sudo netstat -tulpn` on the VM that uses TALK_PORT=1002 ?

**Reply by szaimen** [reply]
can you upload that image again?

**Reply by DrPleb** [reply]
![tlpn](https://github.com/nextcloud/all-in-one/assets/132892195/fe267a52-b877-4762-80a5-c9cb1981f8d3)

**Reply by szaimen** [reply]
So 1002 is correctly opened.

Is port 1002 also added here? 

![image](https://github.com/nextcloud/all-in-one/assets/42591237/2991374d-237f-4616-abbe-b076a29f8f98)

**Reply by DrPleb** [reply]
![tlpn](https://github.com/nextcloud/all-in-one/assets/132892195/80306fda-40fc-46b8-8a6b-0a2bb55c0839)

If I re-run the docker command with port 3478 and change it on the firewall it works...if I choose any other port it doesn't work...

**Reply by szaimen** [reply]
Can you try out a higher port thatn 1024 for a test? Like e.g. 3479

**Reply by DrPleb** [reply]
No way...it works! Is there any technical reason for this?
Thank you so much!
![tlpn](https://github.com/nextcloud/all-in-one/assets/132892195/564a5a79-1e7d-404c-a993-3bde37705123)

**Reply by szaimen** [reply]
I don't know honestly. Possibly some infrastructure in between is blocking ports below 1024...

**Comment by Giggitybyte** [comment]
For those visiting this post: The technical reason is because ports below 1024 are 'privileged', meaning they're reserved for use by the system itself and for standard protocols (e.g. HTTP, FTP, IMAP). 

See here for a bit more detail: https://www.w3.org/Daemon/User/Installation/PrivilegedPorts.html
